src/modules/simulator/simulator.h

```diff
@@ -237,10 +236,24 @@ class Simulator : public ModuleParams
 
 	hrt_abstime _last_sim_timestamp{0};
 	hrt_abstime _last_sitl_timestamp{0};
+	hrt_abstime _last_battery_timestamp{0};
 
-	// Lib used to do the battery calculations.
-	Battery _battery {};
-	battery_status_s _battery_status{};
+	class : public Battery
+	{
+		virtual void updateParams() override
+		{
+			_params.v_empty = 3.5f;
+			_params.v_charged = 4.05f;
+			_params.n_cells = 4;
+			_params.capacity = 10.0f;
+			_params.v_load_drop = 0.0f;
+			_params.r_internal = 0.0f;
+			_params.low_thr = 0.15f;
+			_params.crit_thr = 0.07f;
+			_params.emergen_thr = 0.05f;
+			_params.source = 0;
```

Comment:
I'd still call `Battery::updateParams()` here.

Reply:
Why? In the original version of this, the parameters were not used. They were just hard-coded values like this. If I properly use the parameters, won't that change the behavior of the simulator?

Reply:
Not for now, but in case parameters are used/added in future to the base class.